### PR TITLE
chore(iota): Remove extra address generation in client commands

### PR DESCRIPTION
**Pull request by DaughterOfMars**
# Description of change

Removes an extra address generation in `prompt_if_no_config` when calling `iota client new-address`. This function generates a new address if none exists, but this should be done by the command. Instead, two addresses are generated. This PR fixes this by checking the command variant and skipping the extra address generation for this specific command.

## Before

```
Config file ["/Users/thibault/.iota/iota_config/client.yaml"] doesn't exist, do you want to connect to a Iota Full node server [y/N]?y
Iota Full node server URL (Defaults to Iota Testnet if not specified) : 
Select key scheme to generate keypair (0 for ed25519, 1 for secp256k1, 2: for secp256r1):
0
Keys saved as Bech32.
Generated new keypair and alias for address with scheme "ed25519" [amazing-sunstone: 0xf6dffdc2a6af842bac816180a83d31616d9dc216b8a7882a84bdcef08e6326af]
Secret Recovery Phrase : [clutch solution favorite into excuse own divert adjust attract couple uncle exile]
Keys saved as Bech32.
[warn] Client/Server api version mismatch, client api version: 0.8.0-alpha, server api version: 0.7.3-rc
╭───────────────────────────────────────────────────────────────────────────────────────────────────╮
│ Created new keypair and saved it to keystore.                                                     │
├────────────────┬──────────────────────────────────────────────────────────────────────────────────┤
│ alias          │ peaceful-spodumene                                                               │
│ address        │ 0x4c582c8dee21d59e3a57600e1c866c49414cbfbafad1d1c7973e91b1bc7202d5               │
│ keyScheme      │ ed25519                                                                          │
│ recoveryPhrase │ original unhappy thunder gaze marriage advice raw there vocal holiday seat drama │
╰────────────────┴──────────────────────────────────────────────────────────────────────────────────╯
```

## After

```
Config file ["/Users/chloe/.iota/iota_config/client.yaml"] doesn't exist, do you want to connect to a Iota Full node server [y/N]?y
Iota Full node server URL (Defaults to Iota Testnet if not specified) : 
Keys saved as Bech32.
[warn] Client/Server api version mismatch, client api version: 0.8.0-alpha, server api version: 0.7.3-rc
╭────────────────────────────────────────────────────────────────────────────────────────────────╮
│ Created new keypair and saved it to keystore.                                                  │
├────────────────┬───────────────────────────────────────────────────────────────────────────────┤
│ alias          │ unruffled-zircon                                                              │
│ address        │ 0x1b7242b51f96af36a4483b855092dea80be4af518c05e00995c66096aa2c828b            │
│ keyScheme      │ ed25519                                                                       │
│ recoveryPhrase │ rhythm market elephant bitter room maid ozone include snap sport sleep damage │
╰────────────────┴───────────────────────────────────────────────────────────────────────────────╯
```

### CODE REVIEW

File: Cargo.lock
Comment (thibault-martinez) [inline Cargo.lock:1]: How come this changed?

Comment (DaughterOfMars) [reply Cargo.lock:1]: it just changed in my workspace, no idea why. Figured something must have updated automatically.

File: crates/iota/src/iota_commands.rs [hunk 1]
```diff
@@ -1216,23 +1223,18 @@ async fn prompt_if_no_config(
...
                     scheme.to_string()
                 );
                 println!("Secret Recovery Phrase : [{phrase}]");
-                new_address
-            };
-            let alias = env.alias().clone();
-            IotaClientConfig::new(keystore)
-                .with_envs([env])
-                .with_active_address(active_address)
-                .with_active_env(alias)
```
Comment (thibault-martinez) [inline crates/iota/src/iota_commands.rs:1233]: Removed?

Comment (thibault-martinez) [reply crates/iota/src/iota_commands.rs:1233]: I meant just `with_active_env`

Comment (DaughterOfMars) [reply crates/iota/src/iota_commands.rs:1233]: It's done above instead

Comment (thibault-martinez) [reply crates/iota/src/iota_commands.rs:1233]: Where?

Comment (DaughterOfMars) [reply crates/iota/src/iota_commands.rs:1233]: Oh, I forgot. I removed it because the first env is the default anyway, so it does not matter.

**Review by Thoralf-M — APPROVED** [review]

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**